### PR TITLE
Stellar multicurrency

### PR DESCRIPTION
**Pull request by tobiaschielens**

### CODE REVIEW

File: src/components/AccountCard/AccountCard.html [hunk 1]
```diff
@@ -23,8 +23,8 @@
             </v-btn>
           </v-flex>
           <v-flex class="pr-4 pt-3" v-if="displayAmount">
-            <Balance class="pl-3 pr-2 pt-1" small :balances="account.balances" :humanReadable="$attrs.humanReadable" />
-            <Balance v-if="account.lockedBalances.length" class="pl-3 pr-2 pt-1 mr-2" small locked :balances='[{"balance":account.lockedBalance,"asset_code":"TFT"}]' :humanReadable="$attrs.humanReadable"/>
+            <!-- <Balance class="pl-3 pr-2 pt-1" small :balances="account.balances" :humanReadable="$attrs.humanReadable" /> -->
```
Comment (JonasWijne) [inline src/components/AccountCard/AccountCard.html:26]: should be deleted

File: src/components/PaymentItem/PaymentItem.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
       </v-list-item-subtitle>
       <v-list-item-subtitle class="amount"  style="overflow: visible;">
         {{payment.outgoing ? '-' : '+ ' }} {{ payment.amount | formatBalance }}
-        <sup class="currency font-weight-light text-uppercase">
+        <sup class="currency font-weight-light">
```
Comment (JonasWijne) [inline src/components/PaymentItem/PaymentItem.html:9]: Why is this necessary?

File: src/store/modules/Payments.js [hunk 3]
```diff
@@ -28,9 +29,10 @@ export default {
             const id = payload.id;
 
             const payments = payload.payments.filter(p => {
-                if (p.asset_code !== 'TFT') {
+                console.log(config.currencies)
+                if (!config.currencies[p.asset_code]) {
                     console.log(
-                        'Non TFT transactions not yet supported so will be skipped'
+                        'This currency is not supported'
```
Comment (JonasWijne) [inline src/store/modules/Payments.js:35]: Should say the currency thats not supported (for support reasons).

File: src/views/Details/Details.html [hunk 4]
```diff
@@ -8,11 +8,11 @@ <h2 class="text-capitalize">{{account.name}}
           <span>{{getHumanWalletAddress}}</span>
           <br>
         </v-col>
-        <v-col>
+        <!-- <v-col>
```
Comment (JonasWijne) [inline src/views/Details/Details.html:11]: should be deleted

File: src/views/Details/Details.js [hunk 5]
```diff
@@ -26,6 +26,7 @@ export default {
             selectedPayment: null,
             name: null,
             tab: 0,
+            selectedCurrency: "All",
```
Comment (JonasWijne) [inline src/views/Details/Details.js:29]: Probably better to make "All" ⇒  null

File: src/views/Details/Details.js [hunk 6]
```diff
@@ -121,18 +127,27 @@ export default {
...
             'isPaymentLoading',
+            'currencies'
         ]),
         account() {
             return this.accounts.find(a => a.id === this.id);
         },
         accountPayments() {
-            return this.payments(this.id);
+            return this.payments(this.id).filter(payment => {
+                if(this.selectedCurrency == 'All' || payment.asset_code == this.selectedCurrency){
```
Comment (JonasWijne) [inline src/views/Details/Details.js:137]: When selected currency is null this could be more easily checked e.g. `if(!this.selectedCurrency) {return this.payments(this.id)};  '

File: src/views/Transfer/Transfer.html [hunk 7]
```diff
@@ -38,22 +38,31 @@
             <div style="width: 100%; display: flex;">
               <span class="text-capitalize">{{ data.item.name }}</span>
               <v-spacer></v-spacer>
-              {{ data.item.balances[0].balance | formatBalance }} {{data.item.balances[0].asset_code }}
+              {{data.item.balances.find(x=>x.asset_code == selectedCurrency).balance | formatBalance}}
```
Comment (JonasWijne) [inline src/views/Transfer/Transfer.html:41]: Should be moved to computed or method, too much logic in templates can be confusing

File: src/views/Transfer/Transfer.html [hunk 8]
```diff
@@ -62,13 +71,14 @@
...
       :selectedAccount="selectedAccount"
+      :selectedCurrency="selectedCurrency"
     >
     </FormComponent>
 
     <v-row>
       <v-col>
         <span v-if="$route.query.tab != 'receive'"
-          >Fee: 0.1 TFT</span
+          >Fee: 0.1 {{selectedCurrency}}</span
```
Comment (JonasWijne) [inline src/views/Transfer/Transfer.html:81]: We have to check if its always 0.1

File: src/views/Transfer/Transfer.js [hunk 9]
```diff
@@ -46,10 +47,17 @@ export default {
...
         active() {
             return this.$route.query.tab;
         },
+        availableAccounts(){
+            return this.accounts.filter(account => {
+                return account.balances.find(balance => {
+                    return balance.asset_code === this.selectedCurrency
+                })
+            })
+        }
```
Comment (JonasWijne) [inline src/views/Transfer/Transfer.js:60]: Its straight forward, but maybe add some comments, for future reference.

File: src/views/Transfer/Transfer.js [hunk 10]
```diff
@@ -111,7 +119,8 @@ export default {
                     this.selectedAccount.keyPair,
                     this.formObject.to.address,
                     new Number(this.formObject.amount),
-                    this.formObject.message
+                    this.formObject.message,
+                    this.selectedCurrency
                 );
```
Comment (JonasWijne) [inline src/views/Transfer/Transfer.js:125]: Catch should be split up between buildFundedPaymentTransaction and submitFundedTransaction

File: src/views/Transfer/Transfer.js [hunk 11]
```diff
@@ -122,9 +131,9 @@ export default {
                 this.$flashMessage.info(
                     `Successfully transferred ${this.formObject.amount} to ${this.formObject.to.address}.`
                 );
-            } catch {
+            } catch(e) {
                 //@todo show correct error message for multiple errors eg: "reason": "invalid address"
-
+                console.log(e)
                 this.$flashMessage.error(`Payment failed.`);
```
Comment (JonasWijne) [inline src/views/Transfer/Transfer.js:137]: We should make a better error message based on the error

File: src/views/Transfer/components/formComponent/formComponent.js [hunk 12]
```diff
@@ -24,6 +25,9 @@ export default {
         fee: {
             type: Number,
         },
+        selectedCurrency: {
+            type: String
```
Comment (JonasWijne) [inline src/views/Transfer/components/formComponent/formComponent.js:29]: what if none was given

File: src/components/PaymentItem/PaymentItem.html (hunk 2 above)
Comment (tobiaschielens) [reply src/components/PaymentItem/PaymentItem.html:9]: Because it is FreeTFT

File: src/views/Transfer/Transfer.html (hunk 8 above)
Comment (tobiaschielens) [reply src/views/Transfer/Transfer.html:81]: For now it is 0.1 on all currencies, made issue for later https://github.com/threefoldtech/3bot_wallet/issues/79

File: src/views/Details/Details.js (hunk 5 above)
Comment (tobiaschielens) [reply src/views/Details/Details.js:29]: Just moving  logic to the select, works fine now

File: src/views/Transfer/Transfer.js (hunk 11 above)
Comment (tobiaschielens) [reply src/views/Transfer/Transfer.js:137]: should be different issue, there is a lot to handle there

File: src/views/Transfer/Transfer.js (hunk 10 above)
Comment (tobiaschielens) [reply src/views/Transfer/Transfer.js:125]: hasnt been changed in this pr?
